### PR TITLE
Adding functional tests to test page by page migration use cases

### PR DESCRIPTION
**Pull request by ninaceban**
## Description
For Page-by-Page migration it is important that Web SDK works correctly with both at.js 1.x and at.js 2.x. There are some slight differences that need to be checked because different API's are used.
- Assert same session ID, edge cluster are used for both of the requests interact and delivery API, mbox cookie is saved
1. Web SDK to At.js 1.x
2. Web SDK to At.js 2.x
3. At.js 1.x to Web SDK
4. At.js 2.x to Web SDK
- Assert same Visitor ID and Profile is used across pages ( here we update the profile attribute on page A using one library and fetch on Page B the offer based in profile attribute using another library)
5. Web SDK to At.js 1.x
6. Web SDK to At.js 2.x
7. At.js 1.x to Web SDK
8. At.js 2.x to Web SDK

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Improvement (non-breaking change which does not add functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html) or I'm an Adobe employee.
- [ ] I have made any necessary test changes and all tests pass.
- [ ] I have run the Sandbox successfully.

### CODE REVIEW

File: src/constants/cookieNameKey.js [hunk 1]
```diff
@@ -13,3 +13,4 @@ governing permissions and limitations under the License.
 export const IDENTITY = "identity";
 export const CONSENT = "consent";
 export const CLUSTER = "cluster";
+export const MBOX = "mbox";
```
Comment (jfkhoury) [inline src/constants/cookieNameKey.js:16]: Is this used or going to be used anywhere in the code, other that tests?

**Review by carterworks — APPROVED** [review]

File: src/constants/cookieNameKey.js (hunk 1 above)
Comment (ninaceban) [reply src/constants/cookieNameKey.js]: reverted it, and using the other one used in the code.

File: test/functional/specs/Migration/C8085777.js [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+
+const favoriteColor = "violet-1234";
+const networkLogger = createNetworkLogger();
+const config = compose(orgMainConfigMain, debugEnabled, migrationEnabled, {
+  targetMigrationEnabled: true
+});
+createFixture({
+  title:
+    "C8085777: Use same visitor profile in mixed mode implementation: Update profile attribute using " +
+    "web sdk and fetch offer based on profile attr using at.js 2.x",
```
Comment (jonsnyder) [inline test/functional/specs/Migration/C8085777.js:35]: Nice!

**Review by jonsnyder — CHANGES_REQUESTED** [review]

File: sandbox/public/functional-test/testPageWithAtjs1.html [hunk 3]
```diff
@@ -0,0 +1,19 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <meta charset="utf-8" />
+    <title>Alloy Test Page with at.js 2.x</title>
+    <script src="legacyLibraries/visitorapi.min.js"></script>
```
[the remark this answers is not in the document]
Comment (ninaceban) [reply sandbox/public/functional-test/testPageWithAtjs1.html:6]: In this case we just need the ECID generation logic for at.js, which I don't think gets changed so often. Do we know how often the visitor library is released?

File: test/functional/specs/Migration/C8085774.js [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+    // Check that mboxEdgeCluster cookie is present in the response from Konductor
+    const mboxEdgeClusterCookie = await assertKonductorReturnsCookieAndCookieIsSet(
+      MBOX_EDGE_CLUSTER,
+      stateStorePayload
+    );
+
+    // NAVIGATE to clean page
+    await t.navigateTo(TEST_PAGE_AT_JS_TWO);
+    // get delivery API request adding sleep to make sure the request was triggered
+    await sleep(3000);
```
[the remark this answers is not in the document]
Comment (ninaceban) [reply test/functional/specs/Migration/C8085774.js]: thank you

File: test/functional/specs/Migration/C8085774.js [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+
+    // assert the same cluster is used
+    const cluster = await extractCluster(hostname);
+    await t
+      .expect(`mboxedge${mboxEdgeClusterCookie}`)
+      .eql(
+        cluster,
+        "Cluster ID returned from Target Upstream does not match the cluster ID  used in the path to delivery API"
+      );
+  }
```
[the remark this answers is not in the document]
Comment (ninaceban) [reply test/functional/specs/Migration/C8085774.js:96]: thank you

File: test/functional/specs/Migration/C8085773.js [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+    "interact and delivery API",
+  async () => {
+    // Loaded a page with Alloy
+    await injectAlloyAndSendEvent(config);
+
+    const sendEventRequest = networkLogger.edgeEndpointLogs.requests[0];
+    const requestBody = JSON.parse(sendEventRequest.request.body);
+
+    // Check that targetMigrationEnabled flag is sent in meta
+    await assertTargetMigrationEnabledIsSent(requestBody);
```
[the remark this answers is not in the document]
Comment (ninaceban) [reply test/functional/specs/Migration/C8085773.js]: yes thank you

**Review by jonsnyder — APPROVED** [review]

**Pull request merged**